test/c/test_optimize_1q_decomposition.c

```diff
@@ -0,0 +1,378 @@
+/// cheaper.
+/// @return The generated target instance.
+QkTarget *get_rz_sx_target(void) {
+    QkTarget *target_rz_sx = qk_target_new(1);
+
+    double inst_errors[2] = {0., 2.5e-4};
+    double inst_durations[2] = {0., 5e-9};
+    QkGate gates[2] = {QkGate_RZ, QkGate_SX};
+    // TODO: Update this part to use parameters once we support them.
+    double rz_params[1] = {3.14};
+
+    for (int idx = 0; idx < 2; idx++) {
+        QkTargetEntry *entry;
+        if (gates[idx] == QkGate_RZ) {
+            entry = qk_target_entry_new_fixed(gates[idx], rz_params);
+        } else {
+            entry = qk_target_entry_new(gates[idx]);
+        }
+        uint32_t qargs[1] = {
+            0,
+        };
+        qk_target_entry_add_property(entry, qargs, 1, inst_durations[idx], inst_errors[idx]);
+        qk_target_add_instruction(target_rz_sx, entry);
+    }
+    return target_rz_sx;
+}
+
+/// @brief Generates a target with overcomplete basis, rz is cheaper than ry is cheaper than u.
+/// @return The generated target instance.
+QkTarget *get_rz_ry_u_target(void) {
+    QkTarget *target_rz_ry_u = qk_target_new(1);
+
+    double gate_errors[3] = {1e-4, 2e-4, 5e-4};
+    double gate_durations[3] = {1e-9, 5e-9, 9e-9};
+    QkGate u_gates[3] = {QkGate_RZ, QkGate_RY, QkGate_U};
+    // TODO: Update this part to use parameters once we support them.
+    double rz_params[1] = {3.14};
+    double rx_params[1] = {3.14};
+    double u_params[3] = {3.14, 3.14 / 2., 3.14 / 4.};
+
+    double *gate_params[3] = {rz_params, rx_params, u_params};
+    for (int idx = 0; idx < 3; idx++) {
+        QkTargetEntry *u_entry = qk_target_entry_new_fixed(u_gates[idx], gate_params[idx]);
+        uint32_t qargs[1] = {
+            0,
+        };
+        qk_target_entry_add_property(u_entry, qargs, 1, gate_durations[idx], gate_errors[idx]);
+        qk_target_add_instruction(target_rz_ry_u, u_entry);
+    }
+    return target_rz_ry_u;
+}
+
+/// @brief Generates a target with hadamard and phase, we don't yet have an explicit decomposer
+/// but we can at least recognize circuits that are native for it
+/// @return The generated target instance.
+QkTarget *get_h_p_target(void) {
+    QkTarget *target_h_p = qk_target_new(1);
+
+    double inst_durations[2] = {3e-9, 0.};
+    double inst_errors[2] = {3e-4, 0.};
+    QkGate gates[2] = {QkGate_H, QkGate_Phase};
+    // TODO: Update this part to use parameters once we support them.
+    double phase_params[1] = {3.14};
+
+    for (int idx = 0; idx < 2; idx++) {
+        QkTargetEntry *entry;
+        if (gates[idx] == QkGate_Phase) {
+            entry = qk_target_entry_new_fixed(gates[idx], phase_params);
+        } else {
+            entry = qk_target_entry_new(gates[idx]);
+        }
+        uint32_t qargs[1] = {
+            0,
+        };
+        qk_target_entry_add_property(entry, qargs, 1, inst_durations[idx], inst_errors[idx]);
+        qk_target_add_instruction(target_h_p, entry);
+    }
+    return target_h_p;
+}
```

Comment:
This doesn't really offer much value this is not a 1q universal basis set so we won't ever have a decomposer for it.